### DISCUSSION TITLE
Stacked Pie/Donut Chart

### DISCUSSION THREAD

**Discussion by jakiestfu** [airbnb/visx#1154] [Q&A] [answered]
I'm looking to show multiple datasets within my pie chart. I am able to iterate through my data and render child charts, but the problem I'm having is that they're not actually "stacked" like I'd expect, they're just separate rings independent of the arc prior.

Here is an image of what I'm trying to emulate: 

<img width="421" alt="Screen Shot 2021-04-08 at 5 22 00 PM" src="https://user-images.githubusercontent.com/1041792/114111574-30b75a80-988f-11eb-9c92-b9c8b4864bdd.png">

Thoughts?

**Comment by hshoff** [comment, marked as answer]
Hi @jakiestfu,

You're looking for a [sunburst chart](https://observablehq.com/@d3/zoomable-sunburst). To do this you need to:

1. create a `root` using `hierarchy()`  from either `d3-hierarchy` or `@visx/hierarchy` re-export alias
2. use `<Partition root={root}>{(partition) => // stuff}</Partition>` passing your root as a prop
3. in the `<Partition>` function as children, map over `partition.descendants()` nodes returning an `<Arc />` for each node

more details/an example here: https://github.com/airbnb/visx/discussions/1085

Hope this helps!

**Reply by jakiestfu** [reply]
Thank you so much!

**Reply by jakiestfu** [reply]
@hshoff I'm almost there with the sunburst chart, I'm just having one small issue. It doesn't seem that the subsequent rings are wrapping around fully, for reasons I'm unsure of.

For example, B folder size is 144kb and each file is 48kb, but those three files are not spanning the width of the folder size.

![Screen Shot 2021-04-09 at 5 22 56 PM](https://user-images.githubusercontent.com/1041792/114252136-a3403d00-9958-11eb-9f22-c675d997379b.png)

I have my sum defined correctly I think:

```js
const root =
    hierarchy(folderInfo)
      .sum((data) => data.bytes)
      .sort((a, b) => b.data.bytes - a.data.bytes)
```

but when I look at the `value` property of the parent, it's way larger than `144kb`, probably because it is summing all of the data prior. More D3 related than this, but I feel like I'm missing something.

I think it's possible to solve by fiddling with the parent start and end angles but wasn't sure if there was anything you knew of that I could do. 

![Screen Shot 2021-04-09 at 5 25 22 PM](https://user-images.githubusercontent.com/1041792/114252135-a2a7a680-9958-11eb-860f-e930f25b18db.png)

**Comment by jakiestfu** [comment]
Fixed my issue. In my case, I'm showing directory/file size and I was summing directory sizes which are already sums of file sizes.

<img width="475" alt="Screen Shot 2021-04-09 at 9 10 14 PM" src="https://user-images.githubusercontent.com/1041792/114257935-14431d00-9978-11eb-88bf-94b945cc6d15.png">

Thanks again!
